api/src/pcapi/scripts/titelive_script_2_extract_offer_ids_and_resync.py

```diff
@@ -0,0 +1,136 @@
+import argparse
+import csv
+import datetime
+import logging
+import os
+import socket
+import time
+
+import sqlalchemy as sa
+
+from pcapi import settings
+from pcapi.core import search
+from pcapi.core.offers import models
+import pcapi.core.users.models as users_models
+from pcapi.flask_app import app
+from pcapi.models import db
+from pcapi.models.offer_mixin import OfferValidationType
+from pcapi.repository import repository
+
+
+logger = logging.getLogger(__name__)
+
+HOSTNAME = socket.gethostname()
+
+IN_REJECTED_PRODUCT_IDS_FILE_PATH = "/tmp/OUT_rejected_product_ids.csv"
+OUT_REJECTED_OFFER_IDS_FILE_PATH = "/tmp/OUT_rejected_offer_ids.csv"
+
+
+def reject_inappropriate_offers(productsIds: list[int], dry: bool = False) -> list[str]:
+    offers_query = models.Offer.query.filter(
+        models.Offer.productId.in_(productsIds),
+        models.Offer.validation != models.OfferValidationStatus.REJECTED,
+    )
+
+    offers = offers_query.all()
+
+    if not dry:
+        offers_query.update(
+            values={
+                "validation": models.OfferValidationStatus.REJECTED,
+                "lastValidationDate": datetime.datetime.utcnow(),
+                "lastValidationType": OfferValidationType.CGU_INCOMPATIBLE_PRODUCT,
+            },
+            synchronize_session="fetch",
+        )
+
+    offer_ids = []
+    for offer in offers:
+        offer_ids.append(offer.id)
```

Comment:
Ça fait 2 requêtes SQL (une pour le SELECT, une pour l'UPDATE). On devrait pouvoir faire ça en une seule requête SQL avec un `UPDATE ... RETURNING ...` : 

```suggestion
    res = db.session.execute(
        sqla.update(Offer)
        .returning(Offer.id)
        .where(
            Offer.productId.in_(product_ids),
            Offer.validation != OfferValidationStatus.REJECTED,
        )
        .values(
            validation=OfferValidationStatus.REJECTED,
            lastValidationDate=datetime.datetime.utcnow(),
            lastValidationType=OfferValidationType.CGU_INCOMPATIBLE_PRODUCT,
        ),
        execution_options={"synchronize_session": False},
    )
    offer_ids = [offer_id for offer_id, in res.fetchall()]
```

Reply:
Ok, je conserve les deux, le tiens pour le non dry.